posthog/api/hog_function_template.py

```diff
@@ -107,6 +109,21 @@ def _load_templates(cls):
         ]
         sub_templates = derive_sub_templates(templates=templates)
 
+        template_usage = (
+            HogFunction.objects.filter(type="destination")
+            .values("template_id")
+            .annotate(count=Count("template_id"))
+            .order_by("-count")[:500]
+        )
+
+        popularity_dict = {item["template_id"]: item["count"] for item in template_usage}
+
+        for template in templates:
+            if template.id not in popularity_dict:
+                popularity_dict[template.id] = 0
+
+        templates.sort(key=lambda template: (-popularity_dict[template.id], template.name.lower()))
```

Comment:
I don't think we should do this here. This is the class that is refreshing the templates from Node.js side and should do as little as possible. The ordering is way more to do with the API call (the ViewSet) so should be part of the api viewset for templates.

Reply:
Also means you could get inconsistent results between API requests depending on what that pod had cached in this class at the time.

Reply:
moved the code into the viewset and added a test